scanpipe/pipes/__init__.py

```diff
@@ -254,27 +260,47 @@ def update_or_create_dependency(
     corresponding CodebaseResource for `datafile_path`. This is used in the case
     where Dependency data is imported from a scancode-toolkit scan, where the
     root path segments are not stripped for `datafile_path`.
+    If the dependency is resolved and a resolved package is created, we have the
+    corresponsing package_uid at `resolved_to`.
     """
-    dependency = None
-    dependency_uid = dependency_data.get("dependency_uid")
-
     if ignore_dependency_scope(project, dependency_data):
         return  # Do not create the DiscoveredDependency record.
 
-    if not dependency_uid:
-        dependency_data["dependency_uid"] = uuid.uuid4()
-    else:
-        dependency = project.discovereddependencies.get_or_none(
-            dependency_uid=dependency_uid,
+    dependencies = get_dependencies(project, dependency_data)
+
+    for dependency in dependencies:
+        is_for_new_package = (
+            for_package
+            and dependency.for_package
+            and dependency.for_package != for_package
         )
+        if is_for_new_package:
+            DiscoveredDependency.populate_dependency_uuid(dependency_data)
+            dependency = DiscoveredDependency.create_from_data(
+                project,
+                dependency_data,
+                for_package=for_package,
+                resolved_to_package=resolved_to_package,
+                datafile_resource=datafile_resource,
+                datasource_id=datasource_id,
+                strip_datafile_path_root=strip_datafile_path_root,
+            )
+            break
```

Comment:
Is the `break` intentional here? `break` exits the entire loop while `continue` proceed to the next iteration.

Reply:
Yes, this is intentional to put `break` here. In the case where this dependency data is for a new package, we want to create a new `DiscoveredDependency` object with all the attributes properly populated and exit from this loop, as there is no point continuing anymore.